src/dft/descriptor.cxx

```diff
@@ -26,51 +27,126 @@ namespace detail {
             break;
-        case config_param::FWD_DISTANCE: values_.fwd_dist = va_arg(vl, int64_t); break;
-        case config_param::BWD_DISTANCE: values_.bwd_dist = va_arg(vl, int64_t); break;
+        case config_param::FWD_DISTANCE: values_.fwd_dist = va_arg(vl, std::int64_t); break;
+        case config_param::BWD_DISTANCE: values_.bwd_dist = va_arg(vl, std::int64_t); break;
         case config_param::PLACEMENT: values_.placement = va_arg(vl, config_value); break;
         case config_param::COMPLEX_STORAGE:
             values_.complex_storage = va_arg(vl, config_value);
             break;
+        case config_param::REAL_STORAGE:
+            throw mkl::unimplemented("DFT", "set_value", "Real storage not implemented.");
+            break;
         case config_param::CONJUGATE_EVEN_STORAGE:
             values_.conj_even_storage = va_arg(vl, config_value);
             break;
-        default: err = 1;
+        case config_param::ORDERING:
+            throw mkl::unimplemented("DFT", "set_value", "Ordering not implemented.");
+            break;
+        case config_param::TRANSPOSE:
+            throw mkl::unimplemented("DFT", "set_value", "Transpose not implemented.");
+            break;
+        case config_param::PACKED_FORMAT:
+            throw mkl::unimplemented("DFT", "set_value", "Packed format not implemented.");
+            break;
+        case config_param::COMMIT_STATUS:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        default: throw mkl::invalid_argument("DFT", "set_value", "Invalid config_param argument.");
     }
     va_end(vl);
 }
+
 template <precision prec, domain dom>
 descriptor<prec, dom>::descriptor(std::vector<std::int64_t> dimensions)
         : dimensions_(std::move(dimensions)),
           rank_(dimensions.size()) {
-    // Compute default strides.
-    std::vector<std::int64_t> defaultStrides(rank_, 1);
-    for (int i = rank_ - 1; i < 0; --i) {
-        defaultStrides[i] = defaultStrides[i - 1] * dimensions_[i];
+    if (dimensions_.size() == 0) {
+        throw mkl::invalid_argument("DFT", "descriptor", "Cannot have 0 dimensional DFT.");
+    }
+    for (const auto& dim : dimensions) {
+        if (dim <= 0) {
+            throw mkl::invalid_argument("DFT", "descriptor",
+                                        "Invalid dimension value (negative or 0).");
+        }
+    }
+    // Compute default strides - see MKL C interface developer reference for CCE format.
+    std::vector<std::int64_t> strides(rank_ + 1, 1);
+    // The first variable stide value is different.
+    if (rank_ > 1) {
+        strides[rank_ - 1] = dimensions_[rank_] / 2 + 1;
+    }
+    for (int i = rank_ - 2; i > 0; --i) {
+        strides[i] = strides[i + 1] * dimensions_[i];
+    }
+    strides[0] = 0;
+    // Default for correct strides for forward transform.
+    values_.output_strides = strides;
+    if constexpr (dom == domain::COMPLEX) {
```

Comment:
This is a special case for dft's with forward domain == REAL. I believe from the documentation for the MKL C Interface, that this is not being set by default in there. Not sure if it makes sense to do it by default (e.g. would it still make sense when doing a backward transform)? 

In any case, if we want to set it, shouldn't we be checking for domain::REAL instead of complex?

Reply:
Yes, we should be checking for real. I've corrected this in [a152c39](https://github.com/hjabird/oneMKL/pull/3/commits/a152c39b277981ec04c0d2d8b584178a6c8930b9).

The CCE format is not the default in the MKL C interface, but it is recommended. In the OneAPI spec, CCE format is the only packed format (see [here](https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/dft/config_params/storage_formats.html#onemkl-dft-conjugate-even-storage)) which makes it the default.

Yes, it doesn't make sense if doing a backward transform, but I think that is an unfortunate aspect of the spec. Since we have to choose to defaulting for forward or backward transform, I think forward is the more obvious choice:
* Forward comes before backward
* Most data is time-domain.

Reply:
Revised to match the Intel MKL in [f4ec8e5](https://github.com/hjabird/oneMKL/pull/3/commits/f4ec8e57ad5129244e2f6f84dcc19e9fde8230ad).